sprout/appliances/models.py

```diff
@@ -671,9 +671,24 @@ def prolong_lease(self, time=60):
             appliance.prolong_lease(time=time)
 
     def kill(self):
+        save_lives = not self.fulfilled
         if self.appliances:
             for appliance in self.appliances:
-                Appliance.kill(appliance)
+                # The leased_until is reliable sign of whether the appliance was used
+                # Unless someone was messing with DB ;)
+                if appliance.ready and appliance.leased_until is None and save_lives:
```

Comment:
`save_lives` is already known in the local scope, but `appliance.ready` and `appliance.leased_until` aren't (you have to go through `appliance` to get them. When chaining `and`s, it's best to put the least expensive operands to the left, which in this case is probably `save_lives and appliance.leased_until is None and appliance.ready`, because `save_lives` is known locally, `appliance.leased_until` is identity checking, and `appliance.ready` is none of those.